src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -125,34 +142,33 @@ protected virtual string GetHashAlgorithm(string algorithm)
             switch (algorithm)
             {
                 case SecurityAlgorithms.RsaSha1Signature:
-                    return "SHA1";
+                    return SHA1.Create();
 
                 case SecurityAlgorithms.ECDSA_SHA256:
                 case SecurityAlgorithms.HMAC_SHA256:
                 case SecurityAlgorithms.RSA_SHA256:
                 case SecurityAlgorithms.RsaSha256Signature:
-                    return "SHA256";
+                    return SHA256.Create();
 
                 case SecurityAlgorithms.ECDSA_SHA384:
                 case SecurityAlgorithms.HMAC_SHA384:
                 case SecurityAlgorithms.RSA_SHA384:
                 case SecurityAlgorithms.RsaSha384Signature:
-                    return "SHA384";
+                    return SHA384.Create();
 
                 case SecurityAlgorithms.RsaSha512Signature:
                 case SecurityAlgorithms.RSA_SHA512:
                 case SecurityAlgorithms.ECDSA_SHA512:
                 case SecurityAlgorithms.HMAC_SHA512:
-                    return "SHA512";
+                    return SHA512.Create();
 
                 default:
                     throw new ArgumentOutOfRangeException(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10640, algorithm));
             }
         }
```

Comment:
@bartonjs how will users plug in their own HASH algorithms when validating signatures? The current OM allows for passing a Hashing algorithm. AFAIK FxCore doesn't support CryptoConfig. So I am wondering that since the Crypto provider is responsible for creating the HASH algorithm from a string, how can users can plug a custom algorithm.